packages/flutter_tools/lib/src/runner/flutter_command.dart

```diff
@@ -157,10 +157,12 @@ abstract class FlutterCommand extends Command {
       }
     }
 
-    String error = PackageMap.instance.checkValid();
-    if (error != null) {
-      printError(error);
-      return false;
+    if (!(_usesPubOption && argResults['pub'])) {
```

Comment:
Perhaps add a comment that we only validate the pubspec here if we're not going to run `pub` later on?